common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp

```diff
@@ -268,7 +309,261 @@ bool ap_wlan_hal_nl80211::update_vap_credentials(
     std::list<son::wireless_utils::sBssInfoConf> &bss_info_conf_list,
     const std::string &backhaul_wps_ssid, const std::string &backhaul_wps_passphrase)
 {
-    //TODO Implement #346
+    // The bss_info_conf_list contains list of all VAPs and STAs that have to be created
+    // on the radio. If the list is empty, there should be no VAPs left.
+    // At the moment we are taking a shortcut to make this work for certification setup.
+    // We are updating the hostapd.conf files with the info from the bss_info_conf_list
+    // and sending RELOAD command to hostapd making it update the VAPs.
+    // With that approach there are following limitations:
+    // - all the VAPs are expected to be always present in the config, but have
+    //   start_disabled=1 option if not active
+    // - any configuration change causing the hostapd config files to be re-genrated
+    //   and the hostapd processes restarted will override the purplMesh configuration
+
+    // The hostapd config is stored here. The items order in the header and VAPs sections
+    // is mostly preserved (with the exception of the itemes prplMesh modifies). This helps
+    // to keep the configuration the same and easily see what prplMesh is changing.
+
+    // Load hostapd config for the radio
+
+    LOG(DEBUG) << std::string(__FUNCTION__) << " was called with list size: " << bss_info_conf_list.size();
+
+
+    prplmesh::hostapd::Configuration conf = load_hostapd_config(m_radio_info.iface_name);
+    if (!conf) {
+        LOG(ERROR) << "Autoconfiguration: no hostapd config to apply configuration!";
+        return false;
+    }
+
+    // If a Multi-AP Agent receives an AP-Autoconfiguration WSC message containing one or
+    // more M2, it shall validate each M2 (based on its 1905 AL MAC address) and configure
+    // a BSS on the corresponding radio for each of the M2. If the Multi-AP Agent is currently
+    // operating a BSS with operating parameters that do not completely match any of the M2 in
+    // the received AP-Autoconfiguration WSC message, it shall tear down that BSS.
+
+    // decalre a function for iterating over bss-conf and ap-vaps
+    bool abort          = false;
+    auto configure_func = [&abort, &conf, &bss_info_conf_list, &backhaul_wps_ssid,
+                           &backhaul_wps_passphrase](
+                              const std::string vap,
+                              std::list<son::wireless_utils::sBssInfoConf>::iterator bss_it) {
```

Comment:
```suggestion
                              const std::string vap, decltype(bss_info_conf_list)::iterator bss_it) {
```
But maybe it's just me who finds this easier to read.

Reply:
taking the suggestion as-is resulted with:
`error: decltype evaluates to 'std::__cxx11::list<son::wireless_utils::sBssInfoConf>&', which is not a class or enumeration type`
And the problem is the reference, so ended up with:
`std::remove_reference<decltype(bss_info_conf_list)>::type::iterator`
since C++14 we have [remove_reference_t](https://en.cppreference.com/w/cpp/types/remove_reference) which would make it a bit more readable:
`std::remove_reference_t<decltype(bss_info_conf_list)>::iterator`